lang/attribute/discriminator/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+extern crate proc_macro;
+
+/// A marker attribute used to override the discriminator value that should be used.
+#[proc_macro_attribute]
+pub fn discriminator(
```

Comment:
Can you elaborate on the use case for this macro? When would an anchor program use it?

Reply:
So the plan was that `#[discriminator(x)]` would you allow you to override the discriminator, so that we can test an idl with different discriminator than the standard discriminator created with hashing.

As for a use case,I did imagine a usecase where you can specify a single byte as a discriminator, but this syntax does not allow this yet. This would allow you to just use a single byte rather than 8 bytes for discriminator.

What do you think of:

```
#[discriminator([42])]
fn foo(..)
```
or for two byte discriminators:

```
#[discriminator([42,12])]
fn foo(..)
```

Reply:
If you don't think the attribute is useful, I can create a test using a non-anchor rust contract and a hand-written idl, which is then used from rust and ts.

Reply:
I think this could be interesting. How would you handle potential collisions and/or ambiguity with discriminators of different lengths?